src/framework/circuitopt.hpp

```diff
@@ -24,8 +24,12 @@ namespace AER {
 
 class CircuitOptimization {
 public:
-  virtual void optimize_circuit(Circuit& circ) const = 0;
-  void set_config(const json_t &config);
+  virtual void optimize_circuit(Circuit& circ,
```

Comment:
Here it would be cleaner to just pass in either
```c++
virtual void optimize_circuit(Circuit& circ, const Operations::Opset &opset, OutputData &data) cont
```
since allowed gates, snapshots, ops can all be gotten from the opset (which can be gotten from the state)